### ISSUE TITLE
[Epic] Azure Native v3 release

### ISSUE THREAD

**Issue by mjeffryes** [pulumi/pulumi-azure-native#3363]
## Overview

It's been roughly a year since we last released Azure native. We want to release a new major version (v3), in order to:
1. bumping all the default API versions to the latest reasonable one ([details](https://www.pulumi.com/registry/packages/azure-native/version-guide/)) (consolidating and shrinking the SDK bloat since v2), and
2. implementing fixes and improvements that are breaking changes.

## Key KPIs
1. 0 regressions introduced by this release
2. SDK size should be in the ballpark of v2.0.0
3. Closes ~80% of the issues labeled `3.0`, i.e., ~17 issues, including all blocking ones
4. Adoption: 100% of net-new azure-native users start on v3 after GA
5. Adoption: TODO% of existing users upgrade to v3 within three months

## Key Stakeholders

Engineers: @danielrbradley and @thomas11

Approvers: @mjeffryes

@pulumi/docs to coordinate and help with docs updates

Marketing for promotion

## Key Deliverables

- V2 -> V3 release notes with breaking changes and upgrade instructions
- Official release of V3
- pulumi/examples are updated to v3
- pulumi/templates are updated to v3
- Blog post
- Registry docs for V2 are preserved for some amount of time

### References 📔

- [x] [Project View](https://github.com/orgs/pulumi/projects/159/views/1)
- [ ] PR/FAQ 
- [x] [Design Doc](https://docs.google.com/document/d/1-2OgzNg8sZMEh87QS7lsolKpKFNZOOtJp4iqpINA3jE/edit?pli=1) 
- [ ] Decision Log 

## Basic Process

1. Start implementing the non-breaking issues that can go into master, especially those that influence the automatic part of the major version upgrade
    - [Use the Azure RP API when determining default API versions](https://github.com/pulumi/pulumi-azure-native/issues/2661)
    - [Do not squeeze API versions of resources when the path differs](https://github.com/pulumi/pulumi-azure-native/issues/2813)
1. Create v3 branch
1. In parallel
    - Choose new API versions
      - Run discovery to bump default versions
      - Run schema squeeze to remove older versions which can be safely upgraded.
      - Manually curate version selection
    - Implement epic/project issues

**Comment by 1oglop1** [comment]
@mjeffryes the linked issue is 404 (perhaps it is a private project). Is there any public information about this? Eventually you could unpin the issue to prevent the confusion. 
Thank you!

**Comment by mjeffryes** [comment]
I apologize for the inconvenience @1oglop1! I missed that the new issue was private; we'll flip it around and close that one as the duplicate and work here instead.

**Comment by cleverguy25** [comment]
*[Project] Azure Native V3 status update*
Work on the epic has started for real, with Eron helping out the Azure squad. I (tkappler) haven't actually been participating yet due to customer issues and Insights but hope to join soon.

Two must-have and three should-have items have already landed and work on others is in progress.

No blockers have been encountered so far.

[View all status updates](https://github.com/orgs/pulumi/projects/159?pane=info)

**Comment by meagancojocar** [comment]
Closing this as it shipped! https://www.pulumi.com/blog/azure-native-v3/
